app/src/main/java/com/greenfox/tribesoflagopusandroid/AppModule.java

```diff
@@ -53,22 +52,13 @@ public ObjectManager provideObjectManager(SharedPreferences sharedPreferences, G
 
     @Singleton @Provides
     public LoginService provideLoginService() {
-        if (switchLoginOrMockService == true) {
-            return new LoginService() {
-                @Override
-                public Call<User> loginWithUser(@Field("username") String username, @Field("password") String password) {
-                    return null;
-                }
-            };
+        if (isLoginServiceActive) {
+            return serviceFactory.createRetrofitService(LoginService.class, "https://tribes-of-lagopus.herokuapp.com/");
         }
-        return new MockLoginService();
-    }
-
-    public Boolean getSwitchLoginOrMockService() {
-        return switchLoginOrMockService;
+        return serviceFactory.createMockService(MockLoginService.class, "https://tribes-of-lagopus.herokuapp.com/");
     }
 
     public void setSwitchLoginOrMockService(Boolean switchLoginOrMockService) {
-        this.switchLoginOrMockService = switchLoginOrMockService;
+        this.isLoginServiceActive = switchLoginOrMockService;
```

Comment:
@georgezsiga and we dont want to set this either, this will depend only on the environment not on the behaviour of the user or anything